mne/viz/raw.py

```diff
@@ -628,7 +628,8 @@ def plot_raw_psd(raw, fmin=0, fmax=np.inf, tmin=None, tmax=None, proj=False,
     del ax
     psd_list = list()
     if n_fft is None:
-        tmax = raw.times[-1] if tmax is None else tmax
+        if tmax is None or not np.isfinite(tmax):
+            tmax = raw.times[-1]
```

Comment:
Where did you hit `tmax = np.inf`? Everywhere else we use `tmax=None` to mean "the end" so this suggests that somewhere else in our code / examples / tests we have an error that sets `tmax=np.inf`

Reply:
Ahh I see you've just made it so that `None` is supported, too. This should be fine then

Reply:
tutorials/preprocessing/plot_artifacts_detection.py does it in line 90